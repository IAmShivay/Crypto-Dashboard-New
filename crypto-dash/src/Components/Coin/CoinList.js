import React, { useState, useEffect } from 'react';
import axios from 'axios';
import Coin from './Coin';
import { useSelector } from 'react-redux';

function CoinList() {

    const [coins, setCoins] = useState([]);

    const currencyVal = useSelector((state) => state.CryptoCurrency);

    const searchVal = useSelector((state) => state.search);


    useEffect(() => {
        axios
            .get(
                `https://api.coingecko.com/api/v3/coins/markets?vs_currency=${currencyVal}&order=market_cap_desc&per_page=50&page=1&sparkline=true`
            )
            .then((res) => {
                setCoins(res.data);
                console.log('COIN LIST', res.data);
            })
            .catch((error) => console.log(error));
    }, [currencyVal]);

    const filteredCoins = coins.filter((coin) =>
        coin.name.toLowerCase().includes(searchVal.toLowerCase())
    );

    return (
        <div className="py-5 ">
            <div className="px-4 mb-8 xl:mb-9 lg:mb-6 font-bold text-xl tracking-wider text-stone-600">
                <h1 className='dark:text-gray-50'>Cryptocurrency by market cap</h1>
            </div>

            <div className="h-[44rem] lg:h-[41rem] xl:h-[44rem] overflow-y-scroll scrollbar-hide">
                {filteredCoins.map((coin) => {
                    return (
                        <Coin
                            key={coin.id}
                            name={coin.name}
                            marketcap={coin.market_cap}
                            image={coin.image}
                            priceChange={coin.price_change_percentage_24h}
                            currencyVal={currencyVal}
                        />
                    );
                })}
            </div>
        </div>
    );
}

export default CoinList;
