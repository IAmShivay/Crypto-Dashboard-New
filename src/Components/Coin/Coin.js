import React from "react";

const Coin = ({ name, marketcap, image, priceChange, currencyVal }) => {

  let currencyType;
  if (currencyVal === "usd") {
    currencyType = "$";
  } else if (currencyVal === "inr") {
    currencyType = "Rs";
  } else if (currencyVal === "eur") {
    currencyType = "\u20AC";
  }

  return (

    <div className="group hover:scale-105 hover:shadow-xl hover:shadow-blue-100 dark:hover:shadow-lg dark:hover:shadow-blue-400 transition-all duration-200">
      <div className="flex place-content-between mb-2 lg:mb-[12px] xl:pt-1 border-b-2">
        <div className="px-4 group">
          <div className="coin flex ">
            <img className="w-4 mr-2 self-center" src={image} alt="crypto" />
            <h1 className="dark:text-gray-50 self-center font-semibold text-sm tracking-widest group-hover:text-blue-600 dark:group-hover:text-blue-400 dark:group-hover:delay-100 group-hover:delay-100">
              {name}
            </h1>
          </div>

          <div className="xl:pb-1">
            <p className="dark:text-zinc-300 text-zinc-400 font-semibold text-[10px] tracking-widest mb-[14px] group-hover:text-blue-600  dark:group-hover:text-blue-400 dark:group-hover:delay-100 group-hover:delay-100 ml-6">
              Mkt.Cap {currencyType} {marketcap.toLocaleString()}
            </p>
          </div>
        </div>
        <div className="font-semibold tracking-wider text-xs place-items-center mt-[2px] px-4">
          {priceChange < 0 ? (
            <p className="text-rose-500 dark:text-rose-400">
              <span className="">&#9660;</span>
              {priceChange.toFixed(2)} %
            </p>
          ) : (
            <p className="text-emerald-500 dark:text-emerald-400">
              <span className="mr-2">&#9650;</span>
              {priceChange.toFixed(2)} %
            </p>
          )}
        </div>
      </div>
    </div>
  );
};

export default Coin;
