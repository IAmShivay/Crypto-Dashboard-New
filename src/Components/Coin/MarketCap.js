import React from 'react';
import CoinList from './CoinList';


function MarketCap_EndPoint({CryptoCurrency,search}) {
    return (
        <div className="h-[51.4rem] lg:h-[47.3rem] xl:h-[51.4rem]">
            <CoinList currency={CryptoCurrency} search={search}/>
        </div>
    )
}

export default MarketCap_EndPoint;
