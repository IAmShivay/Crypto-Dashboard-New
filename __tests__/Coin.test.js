import React from 'react';
import Coin from '../src/Components/Coin/Coin';
import renderer from 'react-test-renderer';

describe('Coin', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(
      <Coin name="Bitcoin" marketcap={1000000} image="example.com/bitcoin.png" priceChange={0.5} currencyVal="usd" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});