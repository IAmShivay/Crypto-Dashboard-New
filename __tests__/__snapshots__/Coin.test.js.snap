// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin matches the snapshot 1`] = `
<div
  className="group hover:scale-105 hover:shadow-xl hover:shadow-blue-100 dark:hover:shadow-lg dark:hover:shadow-blue-400 transition-all duration-200"
>
  <div
    className="flex place-content-between mb-2 lg:mb-[12px] xl:pt-1 border-b-2"
  >
    <div
      className="px-4 group"
    >
      <div
        className="coin flex "
      >
        <img
          alt="crypto"
          className="w-4 mr-2 self-center"
          src="example.com/bitcoin.png"
        />
        <h1
          className="dark:text-gray-50 self-center font-semibold text-sm tracking-widest group-hover:text-blue-600 dark:group-hover:text-blue-400 dark:group-hover:delay-100 group-hover:delay-100"
        >
          Bitcoin
        </h1>
      </div>
      <div
        className="xl:pb-1"
      >
        <p
          className="dark:text-zinc-300 text-zinc-400 font-semibold text-[10px] tracking-widest mb-[14px] group-hover:text-blue-600  dark:group-hover:text-blue-400 dark:group-hover:delay-100 group-hover:delay-100 ml-6"
        >
          Mkt.Cap 
          $
           
          1,000,000
        </p>
      </div>
    </div>
    <div
      className="font-semibold tracking-wider text-xs place-items-center mt-[2px] px-4"
    >
      <p
        className="text-emerald-500 dark:text-emerald-400"
      >
        <span
          className="mr-2"
        >
          â–²
        </span>
        0.50
         %
      </p>
    </div>
  </div>
</div>
`;
